R/functions-mgf.R

```diff
@@ -95,54 +185,42 @@ readMgf <- function(f, msLevel = 2L,
 ##' @param mgf `character()` of lines defining a spectrum in mgf
 ##'     format.
 ##'
-##' @param mapping named `character` providing the mapping of MGF fields to
-##'     spectra variable names.
-##'
 ##' @author Laurent Gatto, Johannes Rainer
 ##'
 ##' @importFrom stats setNames
 ##'
 ##' @noRd
-.extract_mgf_spectrum <- function(mgf, mapping) {
+.extract_mgf_spectrum <- function(mgf) {
     ## grep description
     desc.idx <- grep("=", mgf, fixed = TRUE)
     desc <- mgf[desc.idx]
-    spec <- mgf[-desc.idx]
 
-    ms <- do.call(rbind, strsplit(spec, "[[:space:]]+", perl = TRUE))
```

Comment:
turns out the `do.call(rbind` eventually with the needed `mode(ms) <- "double"` below is slightly slower than calling `matrix(as.double(unlist(...`